### PR TITLE
Task 2 : E-commerce website based on React js and Typescript

### PR DESCRIPTION
**Pull request by codal-uchoudhary**

### CODE REVIEW

**Comment by codal-uchoudhary** [comment]
Typescript is implemented  but not in efficient way , i will update soon.

File: Task_2/src/components/Products/ProductPage.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    thumbnail: "",
+    images: [],
+  };
+  const {productId} = useParams();
+  console.log(productId)
+
+  const [data, setData] = useState(Prodcut);
+
+  useEffect(() => {
+    const fetchData = async () => {
```
Comment (codal-shubham) [inline Task_2/src/components/Products/ProductPage.tsx]: Not a correct way to call API. useEffect hook runs after the rendering of the entire UI or component completes. So when we put an API call in it, the API call will start after the complete rendering of UI. Instead use **_useCallback_** 

calling the API inside **_useCallback_** ensures that the function remains stable across re-renders, preventing unnecessary API calls and re-renders unless the dependencies of the **_useCallback_** change

File: Task_2/src/components/Products/ProductPage.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import { BsCurrencyDollar } from "react-icons/bs";
+import { GoStarFill } from "react-icons/go";
+import { FaRegHeart } from "react-icons/fa";
+import { RiShoppingCartLine } from "react-icons/ri";
+import { IoMdShare } from "react-icons/io";
+import ProductGridImage from "./ProductImageGrig";
+import { useParams } from "react-router-dom";
+
+const ProductPage: React.FC = () => {
+  const Prodcut: ProductModel = {
```
Comment (codal-shubham) [inline Task_2/src/components/Products/ProductPage.tsx]: I don't think it's needed @codal-uchoudhary

File: Task_2/src/components/Products/ProductCard.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { AiOutlinePercentage } from "react-icons/ai";
+import { useNavigate } from "react-router-dom";
+
+interface ProductCardProps {
+  productObj: ProductModel;
+}
+
+const ProductCard: React.FC<ProductCardProps> = ({ productObj }) => {
+
+  const productId = productObj.id
```
Comment (codal-shubham) [inline Task_2/src/components/Products/ProductCard.tsx]: use optional chaining for undefined or null checks @codal-uchoudhary

File: Task_2/src/components/Products/ProductList.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import ProductCard from "./ProductCard";
+import { useState, useEffect } from "react";
+
+const ProductList: React.FC = () => {
+  const [data, setData] = useState([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
```
Comment (codal-shubham) [inline Task_2/src/components/Products/ProductList.tsx]: Not a correct way to call API @codal-uchoudhary

**Review by codal-shubham — COMMENTED** [review]
@codal-uchoudhary Check comments added and do the needful changes

File: Task_2/src/components/Products/ProductPage.tsx (hunk 1 above)
Comment (codal-uchoudhary) [reply Task_2/src/components/Products/ProductPage.tsx]: Updated.

File: Task_2/src/components/Products/ProductPage.tsx (hunk 2 above)
Comment (codal-uchoudhary) [reply Task_2/src/components/Products/ProductPage.tsx]: Updated

File: Task_2/src/components/Products/ProductCard.tsx (hunk 3 above)
Comment (codal-uchoudhary) [reply Task_2/src/components/Products/ProductCard.tsx]: Done

File: Task_2/src/components/Products/ProductList.tsx (hunk 4 above)
Comment (codal-uchoudhary) [reply Task_2/src/components/Products/ProductList.tsx]: updated

**Review by codal-shubham — APPROVED** [review]

**Pull request merged**